helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -390,6 +404,9 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
     Map<String, ResourceAssignment> newAssignment = calculateAssignment(clusterData, clusterChanges,
         resourceMap, activeNodes, currentBaseline, currentBestPossibleAssignment);
 
+    // Measure BaselineDivergenceGauge.
+    measureBaselineDivergenceGauge(currentBaseline, newAssignment);
```

Comment:
measure and report. Or just "report" in the method name? This one is more consistent with the other metric emitting code.

Reply:
If the new best possible calculate fails, you will still need to report the divergence between the new baseline and the old "previous" best possible state. So maybe you need to calculate whenever a new baseline created or a new best possible assignment created.

Reply:
Good suggestion.